### PR TITLE
feat(components): [cascader] add `tag-effect` prop

### PR DESCRIPTION
**Pull request by ntnyq**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

This PR adds a prop `tag-effect` to component Cascader.

With it makes it easier to change the tag style.

### CODE REVIEW

File: docs/en-US/component/cascader.md [hunk 1]
```diff
@@ -158,6 +158,7 @@ cascader/panel
 | teleported                          | whether cascader popup is teleported                                                                                                                                             | ^[boolean]                                                    | true    |
 | popper-append-to-body ^(deprecated) | whether to append the popper menu to body. If the positioning of the popper is wrong, you can try to set this prop to false                                                      | ^[boolean]                                                    | true    |
 | tag-type                            | tag type                                                                                                                                                                         | ^[enum]`'success' \| 'info' \| 'warning' \| 'danger'`         | info    |
+| tag-effect                          | tag effect                                                                                                                                                                       | ^[enum]`'light' \| 'dark' \| 'plain'`                         | light   |
```
Comment (btea) [inline docs/en-US/component/cascader.md:161]: Please add a version tag.
```suggestion
| tag-effect  ^(2.7.7)                | tag effect                                                                                                                                                                       | ^[enum]`'light' \| 'dark' \| 'plain'`                         | light   |
```

Comment (ntnyq) [reply docs/en-US/component/cascader.md]: Resolved.

**Review by btea — APPROVED** [review]
Thanks!

**Comment by ntnyq** [comment]
Fixed broken style by overriding `border-color` and `background-color` for tag-effect dark and plain.

[Example](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPG1haW4+XG4gICAgPGRpdiBjbGFzcz1cImRlbW8tcm93XCI+XG4gICAgICA8cD50eXBlPC9wPlxuICAgICAgPGVsLXJhZGlvLWdyb3VwIHYtbW9kZWw9XCJ0YWdUeXBlXCI+XG4gICAgICAgIDxlbC1yYWRpbyB2LWZvcj1cInQgaW4gdGFnVHlwZXNcIiA6a2V5PVwidFwiIDp2YWx1ZT1cInRcIj5cbiAgICAgICAgICB7eyB0IH19XG4gICAgICAgIDwvZWwtcmFkaW8+XG4gICAgICA8L2VsLXJhZGlvLWdyb3VwPlxuICAgIDwvZGl2PlxuICAgIDxkaXYgY2xhc3M9XCJkZW1vLXJvd1wiPlxuICAgICAgPHA+bGlnaHQ8L3A+XG4gICAgICA8ZWwtY2FzY2FkZXJcbiAgICAgICAgdi1tb2RlbD1cImNob2ljZXNcIlxuICAgICAgICA6b3B0aW9ucz1cIm9wdGlvbnNcIlxuICAgICAgICA6cHJvcHM9XCJwcm9wc1wiXG4gICAgICAgIG11bHRpcGxlXG4gICAgICAgIGZpbHRlcmFibGVcbiAgICAgICAgY29sbGFwc2UtdGFnc1xuICAgICAgICA6c2hvdy1hbGwtbGV2ZWxzPVwiZmFsc2VcIlxuICAgICAgICBjb2xsYXBzZS10YWdzLXRvb2x0aXBcbiAgICAgICAgOnRhZy10eXBlPVwidGFnVHlwZVwiXG4gICAgICAgIHRhZy1lZmZlY3Q9XCJsaWdodFwiXG4gICAgICAgIDptYXgtY29sbGFwc2UtdGFncz1cIjFcIlxuICAgICAgLz5cblxuICAgICAgPGVsLXRhZyBlZmZlY3Q9XCJsaWdodFwiIDp0eXBlPVwidGFnVHlwZVwiPiBUYWcgPC9lbC10YWc+XG4gICAgPC9kaXY+XG4gICAgPGRpdiBjbGFzcz1cImRlbW8tcm93XCI+XG4gICAgICA8cD5kYXJrPC9wPlxuICAgICAgPGVsLWNhc2NhZGVyXG4gICAgICAgIHYtbW9kZWw9XCJjaG9pY2VzXCJcbiAgICAgICAgOm9wdGlvbnM9XCJvcHRpb25zXCJcbiAgICAgICAgOnByb3BzPVwicHJvcHNcIlxuICAgICAgICBtdWx0aXBsZVxuICAgICAgICBmaWx0ZXJhYmxlXG4gICAgICAgIGNvbGxhcHNlLXRhZ3NcbiAgICAgICAgOnNob3ctYWxsLWxldmVscz1cImZhbHNlXCJcbiAgICAgICAgY29sbGFwc2UtdGFncy10b29sdGlwXG4gICAgICAgIDp0YWctdHlwZT1cInRhZ1R5cGVcIlxuICAgICAgICB0YWctZWZmZWN0PVwiZGFya1wiXG4gICAgICAgIDptYXgtY29sbGFwc2UtdGFncz1cIjFcIlxuICAgICAgLz5cblxuICAgICAgPGVsLXRhZyBlZmZlY3Q9XCJkYXJrXCIgOnR5cGU9XCJ0YWdUeXBlXCI+IFRhZyA8L2VsLXRhZz5cbiAgICA8L2Rpdj5cbiAgICA8ZGl2IGNsYXNzPVwiZGVtby1yb3dcIj5cbiAgICAgIDxwPnBsYWluPC9wPlxuICAgICAgPGVsLWNhc2NhZGVyXG4gICAgICAgIHYtbW9kZWw9XCJjaG9pY2VzXCJcbiAgICAgICAgOm9wdGlvbnM9XCJvcHRpb25zXCJcbiAgICAgICAgOnByb3BzPVwicHJvcHNcIlxuICAgICAgICBtdWx0aXBsZVxuICAgICAgICBmaWx0ZXJhYmxlXG4gICAgICAgIGNvbGxhcHNlLXRhZ3NcbiAgICAgICAgOnNob3ctYWxsLWxldmVscz1cImZhbHNlXCJcbiAgICAgICAgY29sbGFwc2UtdGFncy10b29sdGlwXG4gICAgICAgIDp0YWctdHlwZT1cInRhZ1R5cGVcIlxuICAgICAgICB0YWctZWZmZWN0PVwicGxhaW5cIlxuICAgICAgICA6bWF4LWNvbGxhcHNlLXRhZ3M9XCIxXCJcbiAgICAgIC8+XG5cbiAgICAgIDxlbC10YWcgZWZmZWN0PVwicGxhaW5cIiA6dHlwZT1cInRhZ1R5cGVcIj4gVGFnIDwvZWwtdGFnPlxuICAgIDwvZGl2PlxuICA8L21haW4+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IHNldHVwIGxhbmc9XCJ0c1wiPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBwcm9wcyA9IHtcbiAgbXVsdGlwbGU6IHRydWUsXG59XG5jb25zdCB0YWdUeXBlcyA9IFsncHJpbWFyeScsICdzdWNjZXNzJywgJ2luZm8nLCAnd2FybmluZycsICdkYW5nZXInXVxuY29uc3Qgb3B0aW9ucyA9IFtcbiAge1xuICAgIHZhbHVlOiAxLFxuICAgIGxhYmVsOiAnQXNpYScsXG4gICAgY2hpbGRyZW46IFtcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6IDIsXG4gICAgICAgIGxhYmVsOiAnQ2hpbmEnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHsgdmFsdWU6IDMsIGxhYmVsOiAnQmVpamluZycgfSxcbiAgICAgICAgICB7IHZhbHVlOiA0LCBsYWJlbDogJ1NoYW5naGFpJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDUsIGxhYmVsOiAnSGFuZ3pob3UnIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogNixcbiAgICAgICAgbGFiZWw6ICdKYXBhbicsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgeyB2YWx1ZTogNywgbGFiZWw6ICdUb2t5bycgfSxcbiAgICAgICAgICB7IHZhbHVlOiA4LCBsYWJlbDogJ09zYWthJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDksIGxhYmVsOiAnS3lvdG8nIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogMTAsXG4gICAgICAgIGxhYmVsOiAnS29yZWEnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHsgdmFsdWU6IDExLCBsYWJlbDogJ1Nlb3VsJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDEyLCBsYWJlbDogJ0J1c2FuJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDEzLCBsYWJlbDogJ1RhZWd1JyB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuICB7XG4gICAgdmFsdWU6IDE0LFxuICAgIGxhYmVsOiAnRXVyb3BlJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICB2YWx1ZTogMTUsXG4gICAgICAgIGxhYmVsOiAnRnJhbmNlJyxcbiAgICAgICAgY2hpbGRyZW46IFtcbiAgICAgICAgICB7IHZhbHVlOiAxNiwgbGFiZWw6ICdQYXJpcycgfSxcbiAgICAgICAgICB7IHZhbHVlOiAxNywgbGFiZWw6ICdNYXJzZWlsbGUnIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMTgsIGxhYmVsOiAnTHlvbicgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAxOSxcbiAgICAgICAgbGFiZWw6ICdVSycsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgeyB2YWx1ZTogMjAsIGxhYmVsOiAnTG9uZG9uJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDIxLCBsYWJlbDogJ0Jpcm1pbmdoYW0nIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMjIsIGxhYmVsOiAnTWFuY2hlc3RlcicgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbl1cblxuY29uc3QgY29sbGFwc2VUYWdzID0gcmVmKGZhbHNlKVxuY29uc3QgdGFnVHlwZSA9IHJlZigncHJpbWFyeScpXG5jb25zdCBjaG9pY2VzID0gcmVmPG51bWJlcltdW10+KFtcbiAgWzEsIDIsIDNdLFxuICBbMSwgMiwgNF0sXG4gIFsxLCAyLCA1XSxcbiAgWzEsIDYsIDddLFxuXSlcbjwvc2NyaXB0PlxuXG48c3R5bGU+XG4uZGVtby1yb3cge1xuICBkaXNwbGF5OiBmbGV4O1xuICBhbGlnbi1pdGVtczogY2VudGVyO1xuICBnYXA6IDIwcHg7XG4gIG1hcmdpbjogMzBweDtcbn1cblxuLmRlbW8tcm93IHAge1xuICBmb250LXNpemU6IDE2cHg7XG4gIG1hcmdpbjogMDtcbiAgY29sb3I6ICM2NjY7XG59XG5cbi5kZW1vLXJvdyAuZWwtY2FzY2FkZXIge1xuICB3aWR0aDogNDQwcHg7XG59XG48L3N0eWxlPlxuIiwic3JjL1BsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xNzQ0My1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCJcbiAgfVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJzcmMvZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIGNvbnN0IHN0eWxlcyA9IFsnaHR0cHM6Ly9wcmV2aWV3LTE3NDQzLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJywgJ2h0dHBzOi8vcHJldmlldy0xNzQ0My1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsKHN0eWxlcylcbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0xNzQ0My1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==)

Still have some style issue when 

- `collapse-tags` is set to false and  `filterable` is set to true 
-  doing a delete tag action
- hit state of tag is active

Any advice?

File: docs/en-US/component/cascader.md [hunk 2]
```diff
@@ -158,6 +158,7 @@ cascader/panel
 | teleported                          | whether cascader popup is teleported                                                                                                                                             | ^[boolean]                                                    | true    |
 | popper-append-to-body ^(deprecated) | whether to append the popper menu to body. If the positioning of the popper is wrong, you can try to set this prop to false                                                      | ^[boolean]                                                    | true    |
 | tag-type                            | tag type                                                                                                                                                                         | ^[enum]`'success' \| 'info' \| 'warning' \| 'danger'`         | info    |
+| tag-effect ^(2.7.7)                 | tag effect                                                                                                                                                                       | ^[enum]`'light' \| 'dark' \| 'plain'`                         | light   |
```
Comment (tolking) [inline docs/en-US/component/cascader.md:161]:
```suggestion
| tag-effect ^(2.7.8)                 | tag effect                                                                                                                                                                       | ^[enum]`'light' \| 'dark' \| 'plain'`                         | light   |
```

**Review by tolking — APPROVED** [review]

**Pull request merged**